### DISCUSSION TITLE
Search breaks when pressing Enter before Ajax call finishes.

### DISCUSSION THREAD

**Discussion by misharigot** [requarks/wiki#3526] [Error / Bug Report]
**Describe the bug**
When typing in the search bar and pressing Enter before the Ajax call that is triggered finishes, the search breaks. After the Ajax call finishes and you click on a search result, you will not be redirected to the page.

**To Reproduce**
1. Click the search bar.
2. Type a search term.
3. Press Enter before the Ajax GET requests finishes.
4. The search is now broken and clicking on a search result will not do anything. 

Additionally, the search results are shown in combination with the text *Type at least two characters to start searching...*. This text is normally only shown when the user has not typed anything in the search bar yet, see image.

**Expected behavior**
Pressing Enter before the Ajax GET request finishes does nothing. Pressing Enter after the Ajax GET request finishes, i.e. when the results are loaded, will navigate to the first page in the search result.

(In my opinion, it might be better for UX to only do the Ajax fetch once the user presses Enter, instead of after typing 2 characters.)

**Screenshots**
![image](https://user-images.githubusercontent.com/15928604/83966152-c9b8c600-a8b8-11ea-9318-c606d6cde809.png)
After typing *test* and immediately pressing enter, the search results break.

**Host Info (please complete the following information):**
 - OS: Docker Container (Linux)
 - Wiki.js version: 2.4.105 (this issue was also present in the previous version, 2.3.x)
 - Database engine: postgres 12.2

**Comment by ghost** [comment]
Guess the same is happening when searching, pressing ESC to close the search preview and when you search that word again it won't give you any results for this term any more.
